File: Amazon/Palindrome/palindrome.go
package main

func IsPalindrome(num int) bool {
	if num < 0 {
		return false
	}

	if num%10 == 0 && num != 0 {
		return false
	}

	if num == Reverse(num) {
		return true
	}

	return false
}

func Reverse(num int) int {
	var reverse int
	for num > 0 {
		reverse = reverse * 10 + num % 10
		num = num / 10
	}
	return reverse
}

File: Amazon/Palindrome/palindrome.js
isPalindrome = function(num) {
    if (num < 0) return false;
    return num === reverse(num);
}  

reverse = function(num) {
    var reversed = 0;
    while (num > 0) {
        reversed = (reversed * 10) + (num % 10);
        num = Math.floor(num / 10);
    }
    return reversed;
}
